pyne/source_sampling.pyx

```diff
@@ -478,14 +477,158 @@ cdef class Sampler:
             rands_proxy = cpp_vector[double](<size_t> rands_size)
             for irands in range(rands_size):
                 rands_proxy[irands] = <double> rands[irands]
-        rtnval = (<cpp_source_sampling.Sampler *> self._inst).particle_birth(rands_proxy)
+        cdef cpp_source_sampling.SourceParticle c_src = (<cpp_source_sampling.Sampler *> self._inst).particle_birth(rands_proxy)
+        return PySourceParticle(c_src.get_x(), c_src.get_y(), c_src.get_z(), c_src.get_e(), c_src.get_w(), c_src.get_c())
+
+
+
+cdef class PySourceParticle:
+    """Constructor for class PySourceParticle
+    
+    Attributes
+    ----------
+    x (double) : The x coordinate of the source particle
+    y (double) : The y coordinate of the source particle
+    z (double) : The z coordinate of the source particle
+    e (double) : The energy of the source particle
+    w (double) : The weight of the source particle
+    c (int) : The cell number of the source particle
+    
+    Methods
+    -------
+    SourceParticle
+    ~SourceParticle
+    get_xyzew
+    get_c
+    
+    Notes
+    -----
+    This class was defined in source_sampling.h
+    
+    The class is found in the "pyne" namespace"""
+
+
+
+    # constuctors
+    def __cinit__(self, double x, double y, double z, double e, double w, int c):
+        self.c_src = cpp_source_sampling.SourceParticle(x, y, z, e, w, c)
+    
+    def get_xyzew(self):
+        """get_xyzew(self)
+        Get source particle x, y, z, e and w
+        
+        Parameters
+        ----------
+        None
+        
+        Returns
+        -------
+        res1 : std::vector< double >
+        
+        """
+        cdef cpp_vector[double] rtnval
+        cdef np.npy_intp rtnval_proxy_shape[1]
+        rtnval = self.c_src.get_xyzew()
         rtnval_proxy_shape[0] = <np.npy_intp> rtnval.size()
         rtnval_proxy = np.PyArray_SimpleNewFromData(1, rtnval_proxy_shape, np.NPY_FLOAT64, &rtnval[0])
         rtnval_proxy = np.PyArray_Copy(rtnval_proxy)
         return rtnval_proxy
+
+    def get_x(self):
```

Comment:
I think that you should use properties here rather than getters for all of these

Reply:
Basically you just need to do, 

```python
@property
def x(self):
    ...
```